cmd/earth/main.go

```diff
@@ -79,7 +81,66 @@ var (
 	GitSha string
 )
 
+// to enable autocomplete, enter
+// complete -o nospace -C "/path/to/earth" earth
+func autoComplete(app *earthApp) {
+	_, found := os.LookupEnv("COMP_LINE")
+	if !found {
+		return
+	}
+
+	compLine := os.Getenv("COMP_LINE")   // full command line
+	compPoint := os.Getenv("COMP_POINT") // where the cursor is
+
+	compPointInt, err := strconv.ParseUint(compPoint, 10, 64)
+	if err != nil {
+		panic(err)
+	}
+
+	flags := []string{}
+	for _, f := range app.cliApp.Flags {
+		for _, n := range f.Names() {
+			if len(n) > 1 {
+				flags = append(flags, n)
+			}
+		}
+	}
+
+	commands := []string{}
+	for _, cmd := range app.cliApp.Commands {
+		commands = append(commands, cmd.Name)
+	}
+
+	potentials, err := autocomplete.GetPotentials(compLine, int(compPointInt), flags, commands)
+	if err != nil {
+		//panic(err) // can't display error or it will show up under tab-completion
```

Comment:
Would stderr work for errors?

I assume that the answer is "no" given that you probably tried `panic`.

Reply:
unfortunately not; if the panic is there it'll suggestion the output as options.